firecloud_project.py

```diff
@@ -177,6 +180,39 @@ def create_private_google_access_dns_zone(context):
     }
   }]
 
+def create_cloud_nats(context):
+  """Creates cloud router resources with automatic NAT enabled.
+
+  Args:
+  context: the DM context object.
+
+  Returns:
+  A resource instantiating the cloud_router.py sub-template.
+  """
+  resources = []
+  for region in FIRECLOUD_NETWORK_REGIONS:
```

Comment:
Do you care if Buffer's regions matches the DM's region. If you setup route with the existing regions in DM, do you need a future migration to add the missing regions from here (RBS - DM) and also set the router for them?
Have you considered just add all missing regions here in this PR? Currently the VPC-SC is just used by AoU, and we don't think we care about adding/removing regions.

Reply:
We (in Terra) probably do care. Otherwise, Terra users may request running WDLs on regions where DM has NOT set up subnets.

Still, I'm leaving syncing up the regions defined by RBS and DM as a separate ticket.

In my head for the next ticket I'm picturing an idempotent script (probably bash or maybe java calling CRL) that looks like:
- Select (or hardcode?) list of supported regions from google cloud
- Select list of projects from rawls's MySQL table
- For each project-region combo
  - Create subnet (and router w/ Cloud NAT?) if not exists
  - Expand subnet to /16

NOTE: making sure new subnets don't collide with old subnets is going to be interesting as RBS and DM are assigning different sets of network prefixes. For example `us-central1` is
`10.128.x.x` when assigned by RBS, but `10.158.x.x` when assigned by DM. However, if RBS hasn't actually run and assigned any subnets yet then maybe we can PR to RBS to have it use the same subnet assignments as DM?

Reply:
> We (in Terra) probably do care. Otherwise, Terra users may request running WDLs on regions where DM has NOT set up subnets.
> 
> Still, I'm leaving syncing up the regions defined by RBS and DM as a separate ticket.
> 
> In my head for the next ticket I'm picturing an idempotent script (probably bash or maybe java calling CRL) that looks like:
> 
> * Select (or hardcode?) list of supported regions from google cloud
> * Select list of projects from rawls's MySQL table
> * For each project-region combo
>   
>   * Create subnet (and router w/ Cloud NAT?) if not exists
>   * Expand subnet to /16
> 
> NOTE: making sure new subnets don't collide with old subnets is going to be interesting as RBS and DM are assigning different sets of network prefixes. For example `us-central1` is
> `10.128.x.x` when assigned by RBS, but `10.158.x.x` when assigned by DM. However, if RBS hasn't actually run and assigned any subnets yet then maybe we can PR to RBS to have it use the same subnet assignments as DM?

That SG. Just remind that your change here does not affect Terra at all. The only affected projects are AoU projects. That's why I offered the option to just add regions since AoU doen't not care, and can also make your future migration easier. (those are in `if high_security_network:` block).
So what you did here, and including your plan (select from Google/Rawls, backfill), those only affect AoU projects if you did it now before PPW. 

I did not notice some ip changed in RBS. Maybe Google changed https://cloud.google.com/vpc/docs/vpc#ip-ranges. 
Good catch!

Reply:
Would be helpful to put what you described above and the proposed change in a doc. Then I can help you to tag few AoU folks.